### ISSUE TITLE
Epoch 0 - loss: inf

### ISSUE THREAD

**Issue by michaelhutt** [mozilla/DeepSpeech#1282]
Following the directions in the README.md for training on the Common Voice dataset, I get:
```
W Parameter --validation_step needs to be >0 for early stopping to work
I STARTING Optimization
I Training of Epoch 0 - loss: inf
```

No custom code.
Using ScientificLinux 7 (basically CentOS 7)
Tensorflow installed from upstream (pip, as instructed), version 1.5.0 (v1.5.0-0-g37aa430d84)
Python 2.7.13
CUDA 9.0, cuDNN 7.0.5
GPU: TITAN X (pascal).

This is exactly what I'm doing:
```
git clone https://github.com/mozilla/DeepSpeech.git
cd DeepSpeech
virtualenv --no-site-packages -p `which python` .ds-env
. .ds-env/bin/activate

export LC_ALL=C
pip install --upgrade pip setuptools requests
pip install -r requirements.txt
pip uninstall -y tensorflow
pip install 'tensorflow-gpu==1.5.0'
pip install deepspeech-gpu

python util/taskcluster.py --target native_client --arch gpu
mkdir -p data/CV
bin/import_cv.py ./data/CV
mv data/CV/cv-other-train.csv data/CV/cv-other-train.csv.backup
perl -ne 'if(!/,\r/) { print; }' < data/CV/cv-other-train.csv.backup > data/CV/cv-other-train.csv

./DeepSpeech.py --train_files data/CV/cv-valid-train.csv,data/CV/cv-other-train.csv --dev_files data/CV/cv-valid-dev.csv --test_files data/CV/cv-valid-test.csv --checkpoint_dir=my-model/cv-chkpt --export_dir=my-model/exported/
```

Changing the learning rate to 0.0001 gives the same output.

Maybe not a bug in the implementation - it could be that the steps in the README are not complete/correct?

**Comment by reuben** [comment]
Can you try without including the `cv-other` files (and with the lower learning rate) and see if it performs better?

**Comment by michaelhutt** [comment]
Sure. I'll report back results on Monday.

**Comment by kdavis-mozilla** [comment]
What about these parameters
```
./DeepSpeech.py \
  --train_files data/CV/cv-valid-train.csv \
  --dev_files data/CV/cv-valid-dev.csv \
  --test_files data/CV/cv-valid-test.csv \
  --train_batch_size 12 \
  --dev_batch_size 12 \
  --test_batch_size 12 \
  --epoch 150 \
  --display_step 1 \ 
  --validation_step 1 \ 
  --dropout_rate 0.30 \
  --default_stddev 0.046875 \
  --learning_rate 0.0001 \
  --log_level 0 \ 
  --checkpoint_dir my-model/cv-chkpt \
  --export_dir=my-model/exported
```

**Comment by michaelhutt** [comment]
It's still running, I'll update when it hits Epoch 0.

**Comment by michaelhutt** [comment]
Okay, it looks like leaving out `cv-other` and dropping the learning rate to 0.0001 gets rid of the infinite loss. I had previously lowered the learning rate and still got `inf`, so I'm not sure if just dropping `cv-other` will do it or if both steps are necessary.

For anybody finding this in the future, this is the full process I used:
```
git clone https://github.com/mozilla/DeepSpeech.git
cd DeepSpeech
virtualenv --no-site-packages -p `which python` .ds-env
. .ds-env/bin/activate

export LC_ALL=C
pip install --upgrade pip setuptools requests
pip install -r requirements.txt
pip uninstall -y tensorflow
pip install 'tensorflow-gpu==1.5.0'
pip install deepspeech-gpu

python util/taskcluster.py --target native_client --arch gpu
mkdir -p data/CV
bin/import_cv.py ./data/CV

./DeepSpeech.py --train_files data/CV/cv-valid-train.csv --dev_files data/CV/cv-valid-dev.csv --test_files data/CV/cv-valid-test.csv --checkpoint_dir=my-model/cv-chkpt --export_dir=my-model/exported/ --learning-rate 0.0001
```

I'm looking at about a week per epoch on a Titan X Pascal and dual Xeon E5-2680v3 @ 2.50GHz

**Comment by Franck-Dernoncourt** [comment]
Note that you may have `loss: inf` for a given epoch and a finite loss at the subsequent epoch.

Example as I was also training DeepSpeech on  the Common Voice dataset, keeping the default learning rate:

```
I Training of Epoch 0 - loss: inf
I Training of Epoch 1 - loss: 47.752834
I Training of Epoch 2 - loss: 33.994538
I Training of Epoch 3 - loss: inf
I Training of Epoch 4 - loss: inf
I Training of Epoch 5 - loss: inf
```
